tools/dotcms-cli/cli/src/main/java/com/dotcms/cli/command/files/FilesPush.java

```diff
@@ -58,26 +62,48 @@ public class FilesPush extends AbstractFilesCommand implements Callable<Integer>
     @Override
     public Integer call() throws Exception {
 
-
         // When calling from the global push we should avoid the validation of the unmatched
         // arguments as we may send arguments meant for other push subcommands
         if (!pushMixin.noValidateUnmatchedArguments) {
             // Checking for unmatched arguments
             output.throwIfUnmatchedArguments(spec.commandLine());
         }
 
-        // Getting the workspace
-        var workspace = getWorkspaceDirectory(pushMixin.path());
+        // Make sure the path is within a workspace
```

Comment:
can this logic be moved to a method to make it more readable?

Reply:
is it possible to recreate the issue with an integration test?

Reply:
@nollymar I created a new `resolveWorkspaceAndPath` to handle the logic to validate and resolve the workspace and path as suggested and also new tests were added.